code/src/sixsq/nuvla/server/resources/api_docs/openapi.clj

```diff
@@ -0,0 +1,122 @@
+                       {"application/json"
+                        {:schema {:$ref (str "#/components/schemas/"
+                                             (openapi-schema/resource-schema-name resource-metadata))}}
+                        ;; TODO: generate examples ?
+                        ;; :examples {}
+                        }}
+                  403 {:description "Not authorized"}
+                  404 {:description "The resource was not found"}}}})
+
+
+(defn collection-query
+  "Returns the OpenAPI method spec for the query operation"
+  [{resource-type-name :name :as _resource-metadata}]
+  {:get
+   {:tags        [resource-type-name]
+    :description (str "Returns a collection of " resource-type-name)
+    :responses   {}}})
+
+
+(defn resource-paths
+  "Returns the `paths` section of the OpenAPI spec for the specified resource."
+  [{:keys [type-uri scrud-operations] resource-type-name :name :as resource-metadata}]
+  (let [scrud-op-supported?  (set scrud-operations)
+        resource-param "resource-id"
+        resource-ops   (merge
+                         (when (scrud-op-supported? :get)
+                           (resource-get resource-metadata)))
+        collection-ops (merge
+                         (when (scrud-op-supported? :query)
+                           (collection-query resource-metadata)))]
+    (merge
+      (when (seq collection-ops)
+        {(str "/" type-uri) collection-ops})
+      (when (seq resource-ops)
+        {(str "/" type-uri "/{" resource-param "}")
+         (merge
+           {:parameters [{:name        resource-param
+                          :in          "path"
+                          :required    true
+                          :description (str "The ID of the " resource-type-name)
+                          :schema      {:type   "string"
+                                        :format "uuid"}}]}
+           resource-ops)}))))
+
+
+(defn paths
+  "Returns the `paths` section of the OpenAPI spec."
+  [resources-metadata]
+  (reduce (fn [m resource-metadata]
+            (merge m (resource-paths resource-metadata)))
+          (sorted-map)
+          resources-metadata))
+
+
+(defn schemas
+  "Returns the `schemas` section of the OpenAPI spec."
+  [resources-metadata]
+  (->> resources-metadata
+       (map openapi-schema/->openapi-schemas)
+       (apply merge)
+       (into (sorted-map))))
+
+
+(defn openapi-spec
+  "Returns the OpenAPI spec for the given resource types."
+  [resources-metadata]
+  (let [nuvla-api-spec {:openapi      "3.0.3"
+                        :info         {:title          "Nuvla API"
+                                       :description    "HTTP-based RESTful API, for the management of Nuvla resources."
+                                       :termsOfService "https://docs.nuvla.io/legal/legal/"
+                                       :contact        {:name  "API Support",
+                                                        :email "apiteam@nuvla.io"}
```

Comment:
Is there an official email address for api support?

Reply:
We have this email for support: `support@sixsq.com`